test/boost/s3_test.cc

```diff
@@ -798,6 +798,66 @@ SEASTAR_THREAD_TEST_CASE(test_chunked_download_data_source_with_delays_proxy) {
     test_chunked_download_data_source(make_proxy_client, 20_MiB);
 }
 
+void do_test_chunked_download_data_source_memory(const client_maker_function& client_maker, size_t object_size) {
+    const sstring base_name(fmt::format("test_object-{}", ::getpid()));
+
+    tmpdir tmp;
+    const auto file_path = tmp.path() / base_name;
+
+    file f = open_file_dma(file_path.native(), open_flags::create | open_flags::wo).get();
+    auto output = make_file_output_stream(std::move(f)).get();
+
+    for (size_t bytes_written = 0; bytes_written < object_size;) {
+        auto rnd = tests::random::get_bytes(std::min(object_size - bytes_written, 1024ul));
+        output.write(reinterpret_cast<char*>(rnd.data()), rnd.size()).get();
+        bytes_written += rnd.size();
+    }
+    output.close().get();
+
+    testlog.info("Make client\n");
+    semaphore mem(1_MiB);
+    auto cln = client_maker(mem);
+    auto close_client = deferred_close(*cln);
+    const auto object_name = fmt::format("/{}/{}", tests::getenv_safe("S3_BUCKET_FOR_TEST"), base_name);
+    auto delete_object = deferred_delete_object(cln, object_name);
+    cln->upload_file(file_path, object_name).get();
+
+    testlog.info("Test client memory exhaust");
+    std::vector<input_stream<char>> clients;
+    for (auto _ : {1, 2, 3}) {
```

Comment:
Let's create at least 16 clients:
```suggestion
 for (auto _ : std::views::repeat(0,16)
```

Reply:
It doesnt matter how many clients you have here, the choke point is [here](https://github.com/scylladb/scylladb/pull/25452/files/03c3561aeb13196c038d3487ddf931f44a713469#diff-29fff134a68462827e145783a13bc1058e37a620750861ed2df38d37100bb3d7R818) , even one client will get stuck, switching to 16 clients will make the test unnecessary long-running.

Reply:
I think that having more clients competing on the same 1MB resource can reveal faster clients that are stack forever here: `auto buf = clients[idx].read().get();`
But I don't have strong opinion on it or any tests either.

Why do you think the test will take longer with more clients?